net-utils/src/lib.rs

```diff
@@ -102,25 +99,38 @@ pub fn get_cluster_shred_version_with_binding(
         .ok_or_else(|| anyhow::anyhow!("IP echo server does not return a shred-version"))
 }
 
+// Limit the maximum number of port verify threads to something reasonable
+// in case the port ranges provided are very large.
+const MAX_PORT_VERIFY_THREADS: usize = 64;
+
 /// Checks if all of the provided TCP/UDP ports are not reachable by the machine at
 /// `ip_echo_server_addr`. Tests must complete within timeout provided.
-/// Tests will run in parallel when possible.
+/// Tests will run concurrently when possible, using up to 64 threads for IO.
 pub fn verify_reachable_ports(
     ip_echo_server_addr: &SocketAddr,
     tcp_listeners: Vec<(u16, TcpListener)>,
     udp_sockets: &[&UdpSocket],
 ) -> bool {
     let rt = tokio::runtime::Builder::new_current_thread()
```

Comment:
Looks suspicious that we create a runtime just to do these checks. Why not take runtime as argument? In this case in tests we could be more explicit about runtime configuration

Reply:
This is public API, mistake like this can not be fixed anymore. At least not easily.

Reply:
This make life more complicated